xdsl/frontend/dialects/arith.py

```diff
@@ -4,86 +4,86 @@
 from xdsl.frontend.dialects.builtin import index, i1, i32, i64, f16, f32, f64
 from xdsl.ir import Operation
 
-_Int = TypeVar("_Int", bound=Union[index, i1, i32, i64], covariant=True)
```

Comment:
I would check with @georgemitenkov  if this is fine or not

Reply:
Feels like we're never going to get subclasses of i1 etc, we could probably revisit if we think it might be possible. It definitely doesn't like passing covariant arguments

Reply:
Okay! Fine for me.
Here, the covariant thing is mostly useful for going from `i1 | i32` to `i1` though, it's not necessary about subclasses of these operations.

Reply:
ah, I didn't think of that!

Reply:
Yep, I think covariance came from cases like `i1 | i32` to `i1` but I do not think this matters too much, I am fine with removing it for now